layer/layer_store.go

```diff
@@ -35,6 +37,24 @@ type layerStore struct {
 	mountL sync.Mutex
 }
 
+// NewStoreFromConfig creates a new Store instance
+func NewStoreFromConfig(root string, driverName string, graphOptions []string, uidMaps, gidMaps []idtools.IDMap) (Store, error) {
```

Comment:
I'm ok with this.

Also, I think that we should leave `NewFSMetadataStore` calls out of here. `root/image/overlay/layerdb` is a docker specific path and should be set by daemon.